canine/localization/nfs.py

```diff
@@ -116,6 +108,20 @@ def localize(self, inputs: typing.Dict[str, typing.Dict[str, str]], patterns: ty
         """
         if overrides is None:
             overrides = {}
+
+        # automatically override inputs that are absolute paths residing on the same
+        # NFS share and are not Canine outputs
+
+        # XXX: this can be potentially slow, since it has to iterate over every
+        #      single input. It would make more sense to do this before the adapter
+        #      converts raw inputs.
+        for input_dict in inputs.values():
+            for k, v in input_dict.items():
+                if k not in overrides:
+                    if re.match(r"^/", v) is not None and self.same_volume(v) and \
+                      re.match(r".*/outputs/\d+/.*", v) is None:
```

Comment:
My check that the input filename is a Canine output is very brittle. Perhaps we should have a better way of tagging whether a directory was generated by Canine?

Reply:
Do we want to move ahead with the output df idea I brought up a couple days ago?

Reply:
So we'd just search for `outputs/.k9df.pickle`? This is definitely more robust than assuming `outputs/\d+` is always a Canine-generated path.

I think this is a separate issue that should be handled in a separate PR.